lib/practice_exercises.rb

```diff
@@ -1,13 +1,30 @@
-
-# Time Complexity: ?
-# Space Complexity: ?
+require 'pry'
+# Time Complexity: O(n)
+# Space Complexity: O(1)
 def is_palindrome(string)
-  raise NotImplementedError, "Not implemented yet"
-end
+  array = string.gsub((/(?i)[^a-z]/),"").downcase.chars
+  array_length = array.length - 1
+  
+  array.each_with_index do |letter,index|
+    return false if letter != array[array_length - index]
+  end
 
-# Time Complexity: ?
-# Space Complexity: ?
-def longest_prefix(strings)
-  raise NotImplementedError, "Not implemented yet"
+  return true
 end
 
+# Time Complexity: O(n^2)
+# Space Complexity: O(1)
+def longest_prefix(strings)
```

Comment:
I would say this is O(n * m) where n is the number of characters in the longest word and n is the number of words.

Otherwise this works!